Add explosion effects to RA bridges

JovialFeline opened the pull request:
Current -> added effects:

https://github.com/user-attachments/assets/df51c3b4-9711-4b50-a1e2-ec58ee8cf8df

For more oomph, bridge damage or destruction in RA '96 spawns a fireball and its associated sound. This PR adds the same effect for ORA bridges, with an added crumbling sound if it's a full collapse.

I'm not confident about the names used, so suggestions for them are extra welcome. Originally, the shore-connecting pieces of the long bridge (`BR1` and `BR2`) do not share this behavior, but I didn't see the harm in adding it.

RA1 code references:
- `Destroy_Bridge_At`, where `ANIM_NAPALM3` can be spawned: https://github.com/electronicarts/CnC_Red_Alert/blob/0dc09bb1d6188d1ec79ab4765b22a48b75ffee32/CODE/MAP.CPP#L1790
- `ANIM_NAPALM3` data: https://github.com/electronicarts/CnC_Red_Alert/blob/0dc09bb1d6188d1ec79ab4765b22a48b75ffee32/CODE/ADATA.CPP#L769

PunkPun reviewed (commented):
It seems the explostion happens twice. Once when bridge entered a damage state, and another time when it's actually destroyed. 

Also the long pieses of bridge not exploding looks a bit weird maybe?

JovialFeline committed:
Add explosion effects to RA bridges

JovialFeline commented:
I went by the original behavior to have both explosions. Each "death" of the bridge spawned the fireball and replaced the template used.

https://github.com/user-attachments/assets/d2bf41c9-8626-4384-b65b-bdfdea5fc7ca

I've now added the explosion to the `BR3` pieces.

PunkPun reviewed (approved):
LGTM

pchote reviewed (approved):
Big polish win, nice work 👍

pchote commented:
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/7d88f5e4339786035848efc0f29d05c847863218
